3rdparty/mshadow/mshadow/dot_engine-inl.h

```diff
@@ -421,12 +421,9 @@ struct BLASEngine<cpu, double> {
   CBLAS_TRANSPOSE p_transa[GROUP_SIZE] = {cblas_a_trans};
   CBLAS_TRANSPOSE p_transb[GROUP_SIZE] = {cblas_b_trans};
 
-  std::vector<const double*> pp_A;
-  std::vector<const double*> pp_B;
-  std::vector<double*> pp_C;
-  pp_A.reserve(batch_count);
-  pp_B.reserve(batch_count);
-  pp_C.reserve(batch_count);
+  std::vector<const double*> pp_A(batch_count, nullptr);
```

Comment:
Thank you for the fix!
It will be better to use `std::array`, since `std::vector` will allocate the memory, which is greater than `batch_count`.

Reply:
I did this because it's precisely the same logic used by the earlier commit.

That being said, I believe the STL guarantees that non-class types will be zero initialized when using the constructor form you propose, so I would expect the performance of the two forms to be identical.